tesseract_core/runtime/tree_transforms.py

```diff
@@ -133,3 +133,36 @@ def filtered_func(new_inputs: dict) -> dict:
         return flatten_with_paths(func(updated_inputs), output_paths)
 
     return filtered_func
+
+
+def filter_pos_func(
+    func: Callable[[dict], dict],
+    default_inputs: dict,
+    output_paths: set[str],
+    keys: list[str],
```

Comment:
I think it makes more sense to have these in the opposite and labelled input_paths
```suggestion
    input_paths: set[str],
    output_paths: set[str],
```

Reply:
sounds good